## **PANDAS LAB 2** (apple_store)

1. Importe el paquete Pandas con el alias pd y Numpy con el alias np.

In [1]:
import numpy as np
import pandas as pd

2. Defina una variable llamada path que contenga la ruta al archivo csv que ha descargado.

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
path = "/content/drive/MyDrive/Colab_Notebooks/apple_store.csv"

3. Usando el método Pandas `read_csv()`, importa el fichero csv en la ruta anterior.

Asigna el valor devuelto a una variable llamada `data`.

Nota: puede omitir el parámetro `sep` para `read_csv()` porque el fichero csv utiliza el separador por defecto ",".

In [6]:
data = pd.read_csv(path)

4. Imprima las 5 primeras filas de data para ver el aspecto de los datos.

Un analista de datos suele hacer esto para tener una idea general de cómo son los datos antes de profundizar en ellos.

In [7]:
data.head()

id                                         track_name  size_bytes  \
0  281656475                                    PAC-MAN Premium   100788224   
1  281796108                          Evernote - stay organized   158578688   
2  281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   100524032   
3  282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   128512000   
4  282935706                                              Bible    92774400   

   price  rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
0   3.99             21292                26          4.0              4.5   
1   0.00            161065                26          4.0              3.5   
2   0.00            188583              2822          3.5              4.5   
3   0.00            262241               649          4.0              4.5   
4   0.00            985920              5320          4.5              5.0   

    prime_genre  
0         Games  
1  Productivity  
2       Weather  
3      Shopping  
4     Reference

5. Imprime el resumen (info) de los datos.

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   id                7197 non-null   int64  
 1   track_name        7197 non-null   object 
 2   size_bytes        7197 non-null   int64  
 3   price             7197 non-null   float64
 4   rating_count_tot  7197 non-null   int64  
 5   rating_count_ver  7197 non-null   int64  
 6   user_rating       7197 non-null   float64
 7   user_rating_ver   7197 non-null   float64
 8   prime_genre       7197 non-null   object 
dtypes: float64(3), int64(4), object(2)
memory usage: 506.2+ KB


6. Imprime el número de columnas de los datos.

In [13]:
data.shape[1] # .shape[0] representa el número de filas y .shape[1] representa el número de columnas

9

7. Imprime todos los nombres de las columnas.

In [11]:
data.columns

Index(['id', 'track_name', 'size_bytes', 'price', 'rating_count_tot',
       'rating_count_ver', 'user_rating', 'user_rating_ver', 'prime_genre'],
      dtype='object')

**Ahora que tenemos una comprensión general de los datos, vamos a empezar a trabajar en las preguntas del desafío.**

8. ¿Cuántas aplicaciones hay en la fuente de datos?

Imprime el # de observaciones de los datos.

*Pista: Tu código debería devolver el número 7197.*

In [14]:
data.shape[0]

7197

9. ¿Cuál es la puntuación media de todas las aplicaciones?

Primero, lee la columna user_rating en una varialbe llamada user_rating.

In [17]:
user_rating = data['user_rating']

Now you can calculate the average of the user_rating data.

Your code should return 3.526955675976101

In [18]:
user_rating.mean()

3.526955675976101

10. ¿Cuántas aplicaciones tienen una valoración media no inferior a 4?

En primer lugar, filtre user_rating cuando su valor sea >= 4.

Asigna el dataframe filtrado a una nueva variable llamada user_rating_high.

In [19]:
user_rating_high = data[data['user_rating'] >= 4]

Ahora obtenga la longitud de user_rating_high que debería devolver 4781.

In [20]:
len(user_rating_high)

4781

11. ¿Cuántos géneros hay en total para todas las aplicaciones?

Define una nueva variable llamada genres que contenga la columna prime_genre de data. Busca en Google cómo obtener valores únicos de una columna del dataframe.

In [21]:
genres = data['prime_genre'].unique()

Imprime la longitud de los valores únicos de genres. Tu código debería devolver 23.

In [22]:
len(genres)

23

12. ¿Cuáles son los 3 géneros con mayor número de aplicaciones?

Lo que quieres hacer es contar el número de apariciones de cada uno de los valores únicos de género. Como ya sabes cómo obtener los valores únicos de género, puedes contar el número de aplicaciones de cada género una a una. Sin embargo, Pandas tiene una función conveniente que le permite contar todos los valores de una columna de marco de datos con un solo comando. Busca en Google "pandas count values" para encontrar la solución. Tu código debería devolver lo siguiente



```
Juegos 3862
Entretenimiento 535
Educación 453
Nombre: prime_genre, dtype: int64
```



In [37]:
genre_occurrences = data['prime_genre'].value_counts()
genre_occurrences.head(3)

prime_genre
Games            3862
Entertainment     535
Education         453
Name: count, dtype: int64

13. ¿Qué género tiene más probabilidades de contener aplicaciones gratuitas?

Primero, filtra data donde el precio sea 0.00. Asigna los datos filtrados a una nueva variable llamada free_apps. A continuación, cuenta los valores de free_apps. Tu código debería devolver:

```
Juegos 2257
Entretenimiento 334
Foto y vídeo 167
Redes sociales 143
Educación 132
Compras 121
Servicios públicos 109
Estilo de vida 94
Finanzas 84
Deportes 79
Salud y bienestar 76
Música 67
Libros 66
Productividad 62
Noticias 58
Viajes 56
Comida y bebida 43
Tiempo 31
Navegación 20
Referencias 20
Negocios 20
Catálogos 9
Medicina 8
Nombre: prime_genre, dtype: int64
```

In [25]:
free_apps = data[data['price'] == 0.00]['prime_genre']
free_apps.value_counts()

prime_genre
Games                2257
Entertainment         334
Photo & Video         167
Social Networking     143
Education             132
Shopping              121
Utilities             109
Lifestyle              94
Finance                84
Sports                 79
Health & Fitness       76
Music                  67
Book                   66
Productivity           62
News                   58
Travel                 56
Food & Drink           43
Weather                31
Business               20
Reference              20
Navigation             20
Catalogs                9
Medical                 8
Name: count, dtype: int64

14. Ahora puedes calcular la proporción de las apps gratuitas de cada género basándote en los recuentos de valores que has obtenido en los dos pasos anteriores.

Desafíate a ti mismo consiguiéndolo con una línea de código. La salida debería ser como:

```
Compras 0,991803
Catálogos 0,900000
Redes sociales 0,856287
Finanzas 0,807692
Noticias 0,773333
Deportes 0,692982
Viajes 0,691358
Comida y bebida 0,682540
Estilo de vida 0,652778
Ocio 0,624299
Libros 0,589286
Juegos 0,584412
Música 0,485507
Foto y vídeo 0,478510
Utilidades 0,439516
Navegación 0,434783
Tiempo 0,430556
Salud y forma física 0,422222
Negocios 0,350877
Productividad 0,348315
Medicina 0,347826
Referencia 0,312500
Educación 0,291391
Nombre: prime_genre, dtype: float64
```

Los números son interesantes, ¿verdad?

In [29]:
prop_free_apps = free_apps.value_counts()/genre_occurrences
prop_free_apps

prime_genre
Book                 0.589286
Business             0.350877
Catalogs             0.900000
Education            0.291391
Entertainment        0.624299
Finance              0.807692
Food & Drink         0.682540
Games                0.584412
Health & Fitness     0.422222
Lifestyle            0.652778
Medical              0.347826
Music                0.485507
Navigation           0.434783
News                 0.773333
Photo & Video        0.478510
Productivity         0.348315
Reference            0.312500
Shopping             0.991803
Social Networking    0.856287
Sports               0.692982
Travel               0.691358
Utilities            0.439516
Weather              0.430556
Name: count, dtype: float64


15. Si un desarrollador intenta ganar dinero desarrollando y vendiendo aplicaciones para la Apple Store, ¿en qué género debería desarrollar las aplicaciones? Suponga que el desarrollo de todas las aplicaciones cuesta el mismo tiempo y dinero.

Esta pregunta se la dejamos a usted. Hay varias formas de resolverla. Lo ideal sería que el resultado fuera como el siguiente:

```
    precio_medio género
21 8.776087 Medicina
11 5,116316 Negocios
4 4,836875 Referencia
6 4,835435 Música
1 4,330562 Productividad
15 4,124783 Navegación
16 4,028234 Educación
12 1,916444 Salud y forma física
20 1,790536 Libros
7 1,647621 Servicios públicos
2 1,605417 Meteorología
18 1,552381 Comida y bebida
14 1,473295 Fotografía y vídeo
0 1,432923 Juegos
8 1,120370 Viajes
10 0,953070 Deportes
13 0,889701 Entretenimiento
17 0,885417 Estilo de vida
22 0,799000 Catálogos
19 0,517733 Noticias
5 0,421154 Finanzas
9 0,339880 Redes sociales
3 0,016311 Compras
```

In [41]:
# 1. Filtrar el DataFrame para obtener solo las aplicaciones de pago:
paid_apps = data[data['price'] > 0.00]

# 2. Calcular el precio medio por género para las aplicaciones de pago:
mean_price_by_genre = paid_apps.groupby('prime_genre')['price'].mean()

# 3. Ordenar los géneros por el precio medio de manera descendente:
mean_price_by_genre_sorted = mean_price_by_genre.sort_values(ascending=False)
mean_price_by_genre_sorted

prime_genre
Medical              13.456667
Music                 9.398451
Catalogs              7.990000
Business              7.881892
Navigation            7.297692
Reference             7.035455
Productivity          6.645172
Education             5.684704
Food & Drink          4.890000
Book                  4.359565
Travel                3.630000
Games                 3.447944
Health & Fitness      3.316923
Sports                3.104286
Utilities             2.939640
Photo & Video         2.825165
Weather               2.819268
Lifestyle             2.550000
Entertainment         2.368109
Social Networking     2.365000
News                  2.284118
Finance               2.190000
Shopping              1.990000
Name: price, dtype: float64